pkg/cmd/server/admin/overwrite_bootstrappolicy.go

```diff
@@ -0,0 +1,153 @@
+				return
+			}
+
+			if err := options.OverwriteBootstrapPolicy(); err != nil {
+				glog.Fatal(err)
+			}
+		},
+	}
+
+	flags := cmd.Flags()
+
+	flags.StringVar(&options.File, "filename", DefaultPolicyFile, "The policy template file containing roles and bindings.")
+	flags.StringVar(&options.MasterConfigFile, "master-config", "master.yaml", "Location of the master configuration file to run from in order to connect to etcd and directly modify the policy.")
+
+	return cmd
+}
+
+func (o OverwriteBootstrapPolicyOptions) Validate(args []string) error {
+	if len(args) != 0 {
+		return errors.New("no arguments are supported")
+	}
+	if len(o.File) == 0 {
+		return errors.New("filename must be provided")
+	}
+	if len(o.MasterConfigFile) == 0 {
+		return errors.New("master-config must be provided")
+	}
+
+	return nil
+}
+
+func (o OverwriteBootstrapPolicyOptions) OverwriteBootstrapPolicy() error {
+	masterConfig, err := ReadMasterConfig(o.MasterConfigFile)
+	if err != nil {
+		return err
+	}
+
+	etcdHelper, err := etcd.NewOpenShiftEtcdHelper(masterConfig.EtcdClientInfo.URL)
+	if err != nil {
+		return err
+	}
+
+	return OverwriteBootstrapPolicy(etcdHelper, masterConfig.PolicyConfig.MasterAuthorizationNamespace, o.File)
+}
+
+func OverwriteBootstrapPolicy(etcdHelper tools.EtcdHelper, masterNamespace, policyFile string) error {
+	mapper := cmdclientcmd.ShortcutExpander{kubectl.ShortcutExpander{latest.RESTMapper}}
+	typer := api.Scheme
+	clientMapper := resource.ClientMapperFunc(func(mapping *meta.RESTMapping) (resource.RESTClient, error) {
+		return nil, nil
+	})
+
+	r := resource.NewBuilder(mapper, typer, clientMapper).
+		ContinueOnError().
+		FilenameParam(policyFile).
+		Flatten().
+		Do()
+
+	if r.Err() != nil {
+		return r.Err()
+	}
+
+	registry := authorizationetcd.New(etcdHelper)
+	roleRegistry := roleregistry.NewVirtualRegistry(registry)
+	roleBindingRegistry := rolebindingregistry.NewVirtualRegistry(registry, registry, masterNamespace)
+
+	return r.Visit(func(info *resource.Info) error {
+		template, ok := info.Object.(*templateapi.Template)
+		if !ok {
+			return errors.New("policy must be contained in a template")
+		}
+
+		for _, item := range template.Objects {
+			switch castObject := item.(type) {
+			case *authorizationapi.Role:
+				ctx := api.WithNamespace(api.NewContext(), castObject.Namespace)
+				roleRegistry.DeleteRole(ctx, castObject.Name)
+				if err := roleRegistry.CreateRole(ctx, castObject); err != nil {
+					return err
+				}
+
+			case *authorizationapi.RoleBinding:
+				ctx := api.WithNamespace(api.NewContext(), castObject.Namespace)
+				roleBindingRegistry.DeleteRoleBinding(ctx, castObject.Name)
+				if err := roleBindingRegistry.CreateRoleBinding(ctx, castObject, true); err != nil {
+					return err
+				}
+
+			default:
+				return errors.New("only roles and rolebindings may be created in this mode")
+			}
+		}
+
+		return nil
+	})
+}
+
+func ReadMasterConfig(filename string) (*configapi.MasterConfig, error) {
```

Comment:
> need to resolve paths relative to the file, since the etcd bits will (today?) contain cert file refs

We need both and there's a dependency chain that needs sorting.  I'm not sure want interdependencies between api and latest, v1, etc.